File: кучи.java
public BinaryHeap(boolean isMax)
{
elts = (T[])new Comparable[2];
elts[0] = null;
elts[1] = null;
last = 0;
this.isMax = isMax;
}

class BinomialHeap<T extends Comparable<T>> {
    private List<BinomialTree<T>> trees;
}

public class Fibonacci {
public static long fibonacci(long n) {
if (n == 0) { return 0; }
else if (n == 1) { return 1; }
else { return fibonacci(n - 1) + fibonacci(n - 2); }
}

File: кучи.py
class BinaryHeap:
    def __init__(self):
        self.heap = [ ]
    
    def push(self, item):
        heapq.heappush(self.heap, item)


class BinomialTree:
    def __init__(self, key):
        self.key = key
        self.children = [ ]

#куча Фибоначчи
def fibonacci(n):
  if n in (1, 2):
    return 1
    return fibonacci(n - 1) + fibonacci(n - 2)
print(fibonacci(10))
